src/System.Diagnostics.Process/src/System/Diagnostics/ProcessStartInfo.cs

```diff
@@ -172,5 +213,82 @@ public string WorkingDirectory
             get { return _directory ?? string.Empty; }
             set { _directory = value; }
         }
+
+        public bool ErrorDialog
+        {
+            get { return _errorDialog; }
+            set { _errorDialog = value; }
+        }
+
+        public IntPtr ErrorDialogParentHandle 
+        {
+            get { return _errorDialogParentHandle; }
+            set { _errorDialogParentHandle = value; }
+        }
+
+        [DefaultValueAttribute("")]
+        public string Verb 
+        {
+            get 
+            {
+                if (_verb == null) return string.Empty;
+                return _verb;
+            }
+            set 
+            {
+                _verb = value;
+            }
+        }
+
+        public string[] Verbs {
+            [ResourceExposure(ResourceScope.None)]
+            [ResourceConsumption(ResourceScope.Machine, ResourceScope.Machine)]
+            get 
+            {
+                ArrayList verbs = new ArrayList();
+                RegistryKey key = null;
+                string extension = Path.GetExtension(FileName);
+                try {
+                    if (extension != null && extension.Length > 0) {
+                        key = Registry.ClassesRoot.OpenSubKey(extension);
+                        if (key != null) {
+                            string value = (string)key.GetValue(String.Empty);
+                            key.Dispose();
+                            key = Registry.ClassesRoot.OpenSubKey(value + "\\shell");
+                            if (key != null) {
+                                string[] names = key.GetSubKeyNames();
+                                for (int i = 0; i < names.Length; i++)
```

Comment:
Nit: Use foreach and below use `string.Equals` instead of `string.Compare`, e.g.:

```c#
foreach (string name in names)
{
    if (!string.Equals(name, "new", StringComparison.OrdinalIgnoreCase))
    {
        verbs.Add(name);
    }
}
```